Passes/PrototypeDeferredPipeline/DeferredLighting.azsl

```diff
@@ -0,0 +1,160 @@
+#include <Atom/Features/PBR/Lighting/StandardLighting.azsli>
+#include <Atom/Features/PBR/Lights/Ibl.azsli>
+#include <Atom/Features/PBR/Decals.azsli>
+
+// Copied from ForwardPassOutput
+struct PSOutput
+{
+    float4 m_diffuseColor  : SV_Target0;     //!< RGB = Diffuse Lighting, A = Blend Alpha (for blended surfaces) OR A = special encoding of surfaceScatteringFactor, m_subsurfaceScatteringQuality, o_enableSubsurfaceScattering
+    float4 m_specularColor : SV_Target1;     //!< RGB = Specular Lighting, A = Unused
+    float4 m_albedo : SV_Target2;            //!< RGB = Surface albedo pre-multiplied by other factors that will be multiplied later by diffuse GI, A = specularOcclusion
+    float4 m_specularF0 : SV_Target3;        //!< RGB = Specular F0, A = roughness
+};
+
+// Since the visual results of the deferred pipeline are (in theory) identical to the results
+// of the main pipeline, this option can be enabled to draw a watermark that confirms what
+// you are seeing is actually the deferred pipeline.
+// TODO(DeferredPOC): Make a way to turn this off by default, and only turn it on as needed for testing. Right now we don't have a way to turn it on programmatically.
+option bool o_deferredPipelineWatermark = true;
+
+PSOutput MainPS(VSOutput IN, in uint sampleIndex : SV_SampleIndex)
+{
+    uint2 screenCoords = IN.m_position.xy;
+
+    // ------- Unpack G-Buffers -------
+
+    float zDepth = PassSrg::m_depthStencilTexture.Load(screenCoords, sampleIndex).r;
+    float3 surfacePosWS = WorldPositionFromDepthBufferMS(PassSrg::m_depthStencilTexture, zDepth, IN.m_position.xy).xyz;
+    float3 encodedNormal = PassSrg::m_normal.Load(screenCoords, sampleIndex).xyz;
+    float3 normal = DecodeNormalSignedOctahedron(encodedNormal.xyz);
+    float2 roughnessMetallic = PassSrg::m_roughnessMetal.Load(screenCoords, sampleIndex).rg;
+    float3 baseColor = PassSrg::m_baseColor.Load(screenCoords, sampleIndex).rgb;
+    
+    if(o_deferredPipelineWatermark)
+    {
+        if(abs(IN.m_texCoord.x-0.5) < 0.01 || abs(IN.m_texCoord.y-0.5) < 0.01) 
+        {
+            float t = sin(50 * (IN.m_texCoord.x + IN.m_texCoord.y)) * 0.5 + 0.5;
+            baseColor *= lerp(float3(1,0.2,1), float3(0.5,1,1), t); 
+        }
+    }
+
+    // Copied from Gems\Atom\Feature\Common\Assets\Materials\Pipelines\MainPipeline\ForwardPass.azsli
+
+    // ------- Surface -------
+
+    Surface surface;
+    surface.position = surfacePosWS;
+
+    // TODO(DeferredPOC): We don't actually have the vertex normal, so just use the per-pixel normal for now
+    surface.vertexNormal = normal;
+    surface.normal = normal;
+    surface.roughnessLinear = roughnessMetallic.x;
+    float specularF0Factor = 0.5f; // TODO(DeferredPOC): Consider passing the actual specF0 through a GBuffer
+    surface.SetAlbedoAndSpecularF0(baseColor, specularF0Factor, roughnessMetallic.y);
+    surface.clearCoat.InitializeToZero();
+        
+    surface.CalculateRoughnessA();
```

Comment:
Should be it's own function that returns Surface
Eventually we'll want to have that function in it's own file so we can change the GBuffer layout in a single location without touching other files

Reply:
Yes, I believe I made this file before the big shader refactor, and I can come back and refactor this code later.